https://issues.redhat.com/browse/ACM-27988--fixes for the z-stream: (#8534)
open-cluster-management/rhacm-docs: release_notes/acm_fixed_issues.adoc

See link:../install/upgrade_hub.adoc#upgrading-by-using-the-operator[Upgrading by using the operator] for more information about upgrades.

*Important:* Cluster lifecycle components and features are within the {mce-short}, which is a software operator that enhances cluster fleet management. Release notes for {mce-short}-specific features are found in at link:../clusters/release_notes/mce_release_notes.adoc#mce-release-notes[Release notes for Cluster lifecycle with {mce-short}].

== Errata 2.15.1

* Delivers updates to one or more product container images.

* Fixes a problem where `ConfigurationPolicies` that check status incorrectly report that the policy is `Compliant` when the status field is missing. With this fix, `ConfigurationPolicies` properly report as `NonCompliant` until that status field matches what is defined in the policy.
link:https://issues.redhat.com/browse/ACM-28224[ACM-28224]

> * Corrects the problem with the monitoring config map that duplicated the `AlertmanagerConfigs` when there was a change in secret names for the alertmanager

Reviewer: ```suggestion
* Corrects the problem where the cluster monitoring config map contains duplicated `AlertmanagerConfigs` due to the change in the Alertmanager secret names in {acm-short} 2.15.
```

link:https://issues.redhat.com/browse/ACM-27841[ACM-27841]

* Fixes a blocker that occurred during cluster deprovisioning, causing the `BareMetalHost` deletion to hang due to a Foreground deletion policy deadlock with the `PreprovisioningImage`. 
https://issues.redhat.com/browse/ACM-27747[ACM-27747]

> * Resolves an issue with rebooting baremetal hosts during the import process, which led to the hosts being reprovisioned, causing the `ClusterInstance` status to report `Provisioning in progress`.`ClusterInstance` status now correctly reflects a completed state for imported clusters that are already provisioned.

Reviewer 2: ```suggestion
* Resolves an issue where `ClusterInstance` status incorrectly reported `Provisioning in progress` for clusters imported after it was previously installed using the Agent-Based Installer or the Installer-Provisioned Infrastructure. `ClusterInstance` status now correctly reflects a completed state for imported clusters that are already provisioned.
```

Author: @sakhoury I need to know what ABI and IPI is and have it spelled out please? TY

Reviewer 2: ABI and IPI are methods for installing OpenShift clusters:
* ABI: Agent-Based Installer
* IPI: Installer-Provisioned Infrastructure

link:https://issues.redhat.com/browse/ACM-27639[ACM-27639]

* Fixes an issue where adding the `observability=disabled` label that failed to remove all observability add-on resources for a target managed cluster. With this fix, all observability add-on resources are removed when the label is added.
https://issues.redhat.com/browse/ACM-27632[ACM-27632]

> * Optimizes the watcher controller to fix overconsumption of resources, specifically memory and CPU, which was causing scalability issues.The fix implements a `ReferenceCache` that maps configuration resources to the `ManifestWorks` that reference them.

Reviewer 3: For the second sentence, I would just state that "The fix Optimizes the reconciliation process by implementing a lookup cache. This replaces the previous resource-intensive checks used to verify if updated ConfigMaps or Secrets are utilized by MCOA."

> link:https://issues.redhat.com/browse/ACM-27304[ACM-27304]

Reviewer: ```suggestion
link:https://issues.redhat.com/browse/ACM-27304[ACM-27304]

* Fixes an issue where the hub Alertmanager secrets in the `openshift-monitoring` namespace became too long, and therefore causing statefulset not to start. The Alertmanager secrets are now appended by a partial cluster ID instead of the cluster name.
link:https://issues.redhat.com/browse/ACM-27722[ACM-27722]
```

Added suggestion for another issue we'd like to highlight in these notes.

See link:../install/upgrade_hub.adoc#upgrading-by-using-the-operator[Upgrading by using the operator] for more information about upgrades.

*Important:* Cluster lifecycle components and features are within the {mce-short}, which is a software operator that enhances cluster fleet management. Release notes for {mce-short}-specific features are found in at link:../clusters/release_notes/mce_release_notes.adoc#mce-release-notes[Release notes for Cluster lifecycle with {mce-short}].

== Errata 2.15.1

* Delivers updates to one or more product container images.

* Fixes a problem where `ConfigurationPolicies` that check status incorrectly report that the policy is `Compliant` when the status field is missing. With this fix, `ConfigurationPolicies` properly report as `NonCompliant` until that status field matches what is defined in the policy.
link:https://issues.redhat.com/browse/ACM-28224[ACM-28224]

> * Corrects the problem where the cluster monitoring config map contains duplicated `AlertmanagerConfigs` due to the change in the Alertmanager secret names in {acm-short} 2.15.

Reviewer: /lgtm

link:https://issues.redhat.com/browse/ACM-27841[ACM-27841]

> * Fixes a blocker that occurred during cluster deprovisioning, causing the `BareMetalHost` deletion to hang due to a Foreground deletion policy deadlock with the `PreprovisioningImage`. 

Reviewer 4: Is foreground supposed to be capitalized?

Author: note sure, it was in the Jira...I think we can get away with not capitalizing this.

> * Fixes a blocker that occurred during cluster deprovisioning, causing the `BareMetalHost` deletion to hang due to a Foreground deletion policy deadlock with the `PreprovisioningImage`. 

Author: ```suggestion
* Fixes a blocker that occurred during cluster deprovisioning, causing the `BareMetalHost` deletion to hang due to a foreground deletion policy deadlock with the `PreprovisioningImage`. 
```

https://issues.redhat.com/browse/ACM-27747[ACM-27747]

> * Resolves an issue where `ClusterInstance` status incorrectly reported `Provisioning in progress` for clusters imported after it was previously installed using the Agent-Based Installer or the Installer-Provisioned Infrastructure. `ClusterInstance` status now correctly reflects a completed state for imported clusters that are already provisioned.

Reviewer 4: ```suggestion
* Resolves an issue where `ClusterInstance` status incorrectly reported `Provisioning in progress` for clusters imported after it was previously installed by using the Agent-Based Installer or the Installer-Provisioned Infrastructure. `ClusterInstance` status now correctly reflects a completed state for imported clusters that are already provisioned.
```

link:https://issues.redhat.com/browse/ACM-27639[ACM-27639]

> * Fixes an issue where adding the `observability=disabled` label that failed to remove all observability add-on resources for a target managed cluster. With this fix, all observability add-on resources are removed when the label is added.

Reviewer 4: ```suggestion
* Fixes an issue where adding the `observability=disabled` label failed to remove all observability add-on resources for a target managed cluster. With this fix, all observability add-on resources are removed when the label is added.
```

I think there is an extra "that" here

https://issues.redhat.com/browse/ACM-27632[ACM-27632]

> * Optimizes the watcher controller to fix overconsumption of resources, specifically memory and CPU, which was causing scalability issues. The fix optimizes the reconciliation process by implementing a lookup cache, which replaces the previous resource-intensive checks that are used to verify if updated `ConfigMaps` or `Secrets` are utilized by the `multicluster-observability-addon`.

Reviewer 4: ```suggestion
* Optimizes the watcher controller to fix overconsumption of resources, specifically memory and CPU, which was causing scalability issues. The fix optimizes the reconciliation process by implementing a lookup cache, which replaces the previous resource-intensive checks that are used to verify if updated `ConfigMaps` or `Secrets` are used by the `multicluster-observability-addon`.
```

link:https://issues.redhat.com/browse/ACM-27304[ACM-27304]

> * Fixes an issue where the hub Alertmanager secrets in the `openshift-monitoring` namespace became too long, and therefore causing statefulset not to start. The Alertmanager secrets are now appended by a partial cluster ID instead of the cluster name.

Reviewer: /lgtm
